### PR TITLE
Fix wxPython 4.0, remove F27, and update license

### PR DESCRIPTION
**Pull request by aimylios**
1. As already mentioned in PR #26, the dependencies for wxPython 4.0 need to be fixed.

2. Copr does not support Fedora 27 any more (since yesterday). All the special handling for F27 can thus be removed from the SPEC file.

3. KiCad is now distributed under the AGPLv3 (since October). The license of the package needs to be updated to reflect this change.

### CODE REVIEW

**Comment by nickoe** [comment]
Thank you.

**Comment by stevefalco** [comment]
Thanks - I'll roll those changes into the downstream project.  Initially, I think it makes sense to provide this in rawhide, then after proving it there, we can consider bringing it to f30 and perhaps f29.  I need to get by-in on that.  The UI has changed enough that it might be inappropriate for backporting.

**Comment by aimylios** [comment]
All these changes are only relevant for KiCad 5.1, so there's still plenty of time left to think about the best approach. If 5.1 is available before F30 gets released, it should (in my opinion) be made available for F29 (but preferably without GTK3, Python 3 and wxPython 4). In the end, it's only a minor release and fully backwards compatible (as far as I know). I would also love to see RC1 with Python support go into Rawhide (which will become F30 very soon), but only the code repository has been tagged up to now, and it seems like docs, translation and libraries are not ready yet.

**Comment by nickoe** [comment]
@aimylios I am not quite sure Ibunderstand your reasoning with usimg gtk2. This meas no wxpython on newer systems. I see you know that.  But as is we are not breakimg anyting existing by going to python3 and gtk3. If we do, it is a bug that we should attempt to address before a final release of 5.1.0.

**Comment by stevefalco** [comment]
@nickoe - I agree with that.  I'd rather move to gtk3, as the tests I've run don't show any problems with it.  If we don't get it in now, then it will have to wait for another release cycle on official Fedora.

@aimylios - I'll be interested if your tests show up any lingering issues with gtk3.  I don't see any strange mouse behavior here.

**Comment by aimylios** [comment]
I don't have a problem with switching to GTK3 and enabling Python 3 and wxPython 4 for Fedora 30. That should definitely be done to finally get full scripting support for Fedora users.
The question is, whether Fedora 29 should get KiCad 5.1 at all and which features to enable. I'm less concerned about Python 3, but to my understanding the GTK3 variant has not seen intensive testing yet on Fedora. And to be on the safe side (i.e., to not break existing installations) I would suggest to keep using GTK2 on F29 and wait for the distro upgrade to switch to GTK3. But it's of course up to @stevefalco and the other maintainers to assess the risk associated with this update and decide what to do.

**Comment by stevefalco** [comment]
Thanks for clarifying, @aimylios .  I agree that rawhide and F30 should get KiCad 5.1 with gtk3 etc.  Finally.

I'm reluctant to move F29 to 5.1, because of the risk, and because of the UI changes, but we'll see how that plays out.

**Comment by stevefalco** [comment]
BTW, I have installed a fedora 28 vm, and I'll try a test build to see if the nightlies could support gtk3 etc on f28.  I'll post the results of that, hopefully later today.

**Comment by stevefalco** [comment]
I've built KiCad with gtk3 for fedora 28.  When I run it, I get a warning:

```
fedora28$ kicad
13:53:50: Warning: Mismatch between the program and library build versions detected.
The library used 3.0 (wchar_t,compiler with C++ ABI 1012,wx containers,compatible with 2.8),
and your program used 3.0 (wchar_t,compiler with C++ ABI 1013,wx containers,compatible with 2.8).
```

It looks like gcc was upgraded in f28, but wx was never rebuilt using the upgraded compiler.  So when I built kicad, I got the newer compiler, hence the discrepancy.  The fix would be to have fedora rebuild wx with the new compiler, but I'm not sure it is worth it.  FWIW, KiCad seems to run fine with the ABI mismatch.

Also, the current nightly builds have the exact same issue, and I get the exact same warning when running the latest nightly.  Thus, compat-wxGTK3 hasn't been rebuilt with the newer compiler either.

Bottom line: it would not make things worse to switch the nightlies for f28 to gtk3, and it would remove the ugly ifdefs.

If you are agreeable, I'll prepare a pull request to do that.

**Comment by aimylios** [comment]
Well, if we just get this warning but everything works and we already see the same issue with compat-wxGTK3-gtk2, then I see no reason not to switch the nightlies to GTK3 and Python 3 on Fedora 28. Please go ahead and submit a PR.

I still plan to do some more testing with my GTK3 build on F29 and will report back. So far, I did not find any issues. The mouse works fine. Maybe the problem I was experiencing has been fixed in the meantime (either by KiCad or any of its dependencies).
